test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -996,6 +1004,199 @@ TEST_F(ClientLbEnd2endTest, RoundRobinSingleReconnect) {
+
+  grpc_connectivity_state CheckConnectivityLocked(
+      grpc_error** connectivity_error) override {
+    return delegate_->CheckConnectivityLocked(connectivity_error);
+  }
+
+  void HandOffPendingPicksLocked(LoadBalancingPolicy* new_policy) override {
+    delegate_->HandOffPendingPicksLocked(new_policy);
+  }
+
+  void ExitIdleLocked() override{
+    delegate_->ExitIdleLocked();
+  }
+
+  void ResetBackoffLocked() override {
+    delegate_->ResetBackoffLocked();
+  }
+
+  void FillChildRefsForChannelz(
+      grpc_core::channelz::ChildRefsList* child_subchannels,
+      grpc_core::channelz::ChildRefsList* ignored) override {
+    delegate_->FillChildRefsForChannelz(child_subchannels, ignored);
+  }
+
+ protected:
+  void ShutdownLocked() override {
+    // noop
+  }
+  Args args_;
+
+ private:
+  grpc_core::OrphanablePtr<LoadBalancingPolicy> delegate_;
+};
+
+class ClientLbInterceptTrailingMetadataTest : public ClientLbEnd2endTest {
+ protected:
+  void SetUp() override {
+    ClientLbEnd2endTest::SetUp();
+    grpc_core::LoadBalancingPolicyRegistry::Builder::
+        RegisterLoadBalancingPolicyFactory(
+            grpc_core::UniquePtr<grpc_core::LoadBalancingPolicyFactory>(
+                grpc_core::New<InterceptTrailingFactory>(this)));
+  }
+
+  void TearDown() override {
+    ClientLbEnd2endTest::TearDown();
+  }
+
+  class InterceptTrailingLb : public ForwardingLoadBalancingPolicy {
+   public:
+    InterceptTrailingLb(
+        const Args& args,
+        const std::string& delegate_lb_policy_name,
+        ClientLbInterceptTrailingMetadataTest* test)
+        : ForwardingLoadBalancingPolicy(args, delegate_lb_policy_name),
+        test_{test} {
+    }
+
+    bool PickLocked(PickState* pick, grpc_error** error) override {
+      bool ret = ForwardingLoadBalancingPolicy::PickLocked(pick, error);
+      // If these asserts fail, then we will need to add code to
+      // proxy the results to the delegate LB.
+      GPR_ASSERT(pick->recv_trailing_metadata == nullptr);
+      GPR_ASSERT(pick->recv_trailing_metadata_ready == nullptr);
+      // OK to add add callbacks for test
+      GRPC_CLOSURE_INIT(
+          &recv_trailing_metadata_ready_,
+          InterceptTrailingLb::RecordRecvTrailingMetadata,
+          this,
+          grpc_schedule_on_exec_ctx);
+      pick->recv_trailing_metadata_ready = &recv_trailing_metadata_ready_;
+      pick->recv_trailing_metadata = &recv_trailing_metadata_;
+      return ret;
+    }
+
+    static void RecordRecvTrailingMetadata(void* arg, grpc_error* err) {
+      InterceptTrailingLb* lb = static_cast<InterceptTrailingLb*>(arg);
+      GPR_ASSERT(err == GRPC_ERROR_NONE);
+      GPR_ASSERT(lb->recv_trailing_metadata_ != nullptr);
+      // an simple check to make sure the trailing metadata is valid
+      GPR_ASSERT(grpc_get_status_code_from_metadata(
+          lb->recv_trailing_metadata_->idx.named.grpc_status->md) ==
+              grpc_status_code::GRPC_STATUS_OK);
+      GRPC_ERROR_UNREF(err);
```

Comment:
No need for this -- closures don't take ownership of the error that they are passed.

https://github.com/grpc/grpc/blob/master/doc/core/grpc-error.md#ownership-rules